lib/ecmascript.ts

```diff
@@ -120,6 +120,24 @@ const BUILTIN_CALENDAR_IDS = [
   'gregory'
 ];
 
+/*
+ * These functions are all used to help TS understand when variables should change types,
+ * but TS can't or won't infer this automatically. They should be used sparingly, because
+ * if used incorrectly can lead to difficult-to-diagnose problems.
+ */
+/* eslint-disable @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function */
+export function assertIsPlainTime(arg: unknown, justification: string): asserts arg is Temporal.PlainTime {}
```

Comment:
Do we also want to have a generic function for other types?  Many of our type casts use other types.

Reply:
Yes - done.